p2p/dialer.go

```diff
@@ -110,6 +116,16 @@ func (m *dialer) exchangeConfig(consumerID, providerID identity.Identity, servic
 	}
 	defer brokerConn.Close()
 
+	ready := make(chan bool)
```

Comment:
It's better to have empty struct chan for such signals since you don't need it's value. If you have bool channel it's confusing and looks like you care if it's true or false.
```go
ready := make(chan struct{})
```